app/controllers/user.js

```diff
@@ -0,0 +1,20 @@
+const logger = require('../logger'),
+  { createUser } = require('../services/user'),
+  { validateUserData } = require('../utils/user'),
+  { encrypt } = require('../utils/encrypt');
+
+exports.signUp = (req, res) => {
+  const validationErrors = validateUserData(req.body);
+  if (validationErrors.length) {
+    return res.status(400).send(validationErrors);
+  }
+  return encrypt(req.body.password)
+    .then(encryptedPassword => createUser({ ...req.body, password: encryptedPassword }))
```

Comment:
Be careful with spread body and send directly to database, if i send extra fields could be persisted whatever i want